content/lectures/lec03.md

```diff
@@ -8,3 +8,60 @@ draft: false
 ---
 
 Coming soon!
+
+## Demos
+
+Docker's implementation of images and containers is wonderful because it allows us to easily package up and send application code to another machine with strong guarantees that the code will work on the other machine. However, we are still missing a crucial component.
+
+Taking our Docker images and spinning up the containers is, as far as we have seen, a relatively manual process. We have used the Docker CLI to download, build, and run containers. What happens when we have multiple containers we want to run in collaboration (maybe a frontend, backend framework)? If there are specific parameters we want to set for certain containers (mounting a volume, joining a network, etc.)? This is where Docker compose comes into play.
+
+Docker compose allows us to automate the process of installing, building, configuring, and running our Docker containers. You could argue that this is something we could do with a shell script of the commands we would run in the CLI, but you'll quickly see that this is brittle and hard for developers to understand or update. Instead, Docker compose allows us to write YAML files which are readable and clearly outline the configuration of each container.
+
+So, let's jump right into some examples. In lecture, we noted that because Docker uses copy-on-write file systems, when changes are made to the filesystem during Docker runtime, those changes are discarded after the container is closed. Let's see if we can write a Docker compose file to create a container that will mount of a volume so we can save logs made during the runtime of our program.
+
+If you've already the website repository, you can find the code for this demo in `/static/demos/docker_compose_demo` (if you need instructions for cloning the repository, take a look at the Docker demo from last week). Let's quickly look over the biggest new addition: `docker compose.yml`. This is a relatively short file with the contents:
+```
+version: '3'
+services:
+  docker_compose_demo_container:
+    image: docker_compose_demo_image:latest
+    command: node index.js
+    ports:
+      - "8080:8080"
+    volumes:
+      - type: bind
+        source: ./log
+        target: /etc/log
```

Comment:
putting this in `/etc` is a little sus. i'd just make a `/log` directory

Reply:
How about `/dev/null`? Sounds good thanks for pointing this out I'll do `/log`